components/inbox-mail/helper.ts

```diff
@@ -0,0 +1,266 @@
+    }
+    else if (!thread.isAdminSupportThread) {
+        if (thread.latestMail.senderId === currentUser.id) {
+            displayName = thread.latestMail.recipientNames.map(r => `${r.first} ${r.last}`).join(", ");
+            const r = thread.latestMail.recipientNames[0];
+            displayInitials = setInitials(r.first, r.last);
+            setColor(recipientIds[0]);
+        } else {
+            const s = thread.latestMail.senderName;
+            const first = s.first ?? "";
+            const last = s.last ?? "";
+            displayName = `${first} ${last}`.trim();
+            displayInitials = setInitials(first, last);
+            setColor(thread.latestMail.senderId);
+        }
+    } else if (thread.isAdminSupportThread && thread.metaData?.autoGenerated) {
+        if (thread.latestMail.senderId) {
+            displayName = thread.latestMail.recipientNames.map(r => `${r.first} ${r.last}`).join(", ");
+        }
+        else {
+            displayName = `${thread.latestMail.senderName.first} ${thread.latestMail.senderName.last}`;
+            DisplayIcon = IconBug
+            background = "#FFE5E5";
+            text = "#A30000";
+        }
+    } else if (thread.isAdminSupportThread && isAdmin && thread.latestMail.senderId === currentUser.id) {
+        displayName = `${thread.latestMail.senderName.first} ${thread.latestMail.senderName.last}`;
+        displayInitials = setInitials(thread.latestMail.senderName.first, thread.latestMail.senderName.last);
+        setColor(thread.latestMail.senderId);
+
+    } else if (thread.isAdminSupportThread && isAdmin) {
+        if (thread.latestMail.senderId === currentUser.id && thread.latestMail.recipientNames.length === 0) {
+            displayName = `${thread.latestMail.senderName.first} ${thread.latestMail.senderName.last}`;
+        } else if (thread.latestMail.senderId === currentUser.id) {
+            displayName = thread.latestMail.recipientNames.map(r => `${r.first} ${r.last}`).join(", ");
+            const r = thread.latestMail.recipientNames[0];
+            displayInitials = setInitials(r.first, r.last);
+            setColor(recipientIds[0]);
+        } else {
+            const s = thread.latestMail.senderName;
+            displayName = `${s.first} ${s.last}`;
+            displayInitials = setInitials(s.first ?? "", s.last ?? "");
+            setColor(thread.latestMail.senderId ?? thread.createdBy);
+        }
+    } else {
+        if (thread.latestMail.senderId === currentUser.id) {
+            displayName = `${thread.latestMail.senderName.first} ${thread.latestMail.senderName.last}`;
+        } else {
+            displayName = thread.latestMail.recipientNames.map(r => `${r.first} ${r.last}`).join(", ");
+        }
+    }
+
+    return {
+        displayName,
+        displayInitials,
+        background,
+        text,
+        recipientIds: recipientIds,
+        DisplayIcon
+    };
+}
+
+export function formatDisplayTimestamp(createdAt: string): string {
```

Comment:
i would put this in the javascript submodules, we have a date-parser file there, where we store different date/time formatting, maybe we can use this formatting for something else too

Reply:
same for the functions below (date time related)